lib/portlayer/storage/vsphere/toolbox_common.go

```diff
@@ -125,3 +128,34 @@ func isInvalidStateError(err error) bool {
 	}
 	return false
 }
+
+// IsToolBoxConflictErr checks for conflictError for online import
+func IsToolBoxStateChangeErr(err error) bool {
+	// check if error has to do with toolbox state changes
+	if soap.IsSoapFault(err) {
+		switch soap.ToSoapFault(err).VimFault().(type) {
+		case types.InvalidState:
+			return true
+		case types.InvalidPowerState:
+			return true
+		case types.GuestOperationsUnavailable:
+			return true
+		case types.SystemError:
+			return strings.Contains(err.Error(), VIX_E_TOOLS_NOT_RUNNING)
```

Comment:
Just curiou8s, but is there a more granular way to check this error? Do we know the format of this error type so that we could parse the code out? I ask because 3016 is bound to show up in a timestamp or something at some point .. although I don't know how large a variety of `types.SystemError`s we get back.